components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsRules.kt

```diff
@@ -12,21 +12,53 @@ import mozilla.components.feature.sitepermissions.SitePermissionsRules.Action.BL
 /**
  * Indicate how site permissions must behave by permission category.
  */
-data class SitePermissionsRules(
+data class SitePermissionsRules internal constructor(
     val camera: Action,
     val location: Action,
     val notification: Action,
-    val microphone: Action
+    val microphone: Action,
+    val autoplayAudible: Action,
+    val autoplayInaudible: Action
 ) {
+
+    constructor(
+        camera: Action,
+        location: Action,
+        notification: Action,
+        microphone: Action,
+        autoplayAudible: AutoplayAction,
+        autoplayInaudible: AutoplayAction
+    ) : this(
+        camera = camera,
+        location = location,
+        notification = notification,
+        microphone = microphone,
+        autoplayAudible = autoplayAudible.toAction(),
+        autoplayInaudible = autoplayInaudible.toAction()
+    )
+
     enum class Action {
-        BLOCKED, ASK_TO_ALLOW;
+        ALLOWED, BLOCKED, ASK_TO_ALLOW;
 
         fun toStatus(): SitePermissions.Status = when (this) {
+            ALLOWED -> SitePermissions.Status.ALLOWED
             BLOCKED -> SitePermissions.Status.BLOCKED
             ASK_TO_ALLOW -> SitePermissions.Status.NO_DECISION
         }
     }
 
+    /**
+     * Autoplay requests will never prompt the user
+     */
+    enum class AutoplayAction {
```

Comment:
```
  /**
     * Autoplay requests will never prompt the user
     */
```
I think can keep autoplay as an `Action`as in a future follow up we want that autoplay shows an UI in the toolbar instead of a prompt

Reply:
I don't think we can here, because even once we're showing autoplay in the UI, it won't be a prompt.  If it's anything like desktop, it'll be an icon that the user can press instead of a dialog.

Reply:
Although this could probably be renamed to be more general.  It's an action that will never prompt the user, what would be a good name for that?

Reply:
What I meant was that `SitePermissionsRules.ASK_TO_ALLOW`  is lower level API is doesn't imply how we are going to ask to allow(`ASK_TO_ALLOW`), it just says that we need to ask, this will be the job of `SitePermissionsFeature` as it is a higher level (closer to the UI) it will determine which permissions should prompt and which not.

Reply:
I'm OK to leave as it is I just want to minimize the change and reuse the concepts that we already have.